packages/ember/tests/routing/decoupled_basic_test.js

```diff
@@ -326,4 +327,217 @@ class extends ApplicationTestCase {
+    this.router.map(function() {
+      this.route('home', { path: '/' });
+    });
+
+    this.add('route:home', Route.extend({
+      controllerName: 'myController',
+      renderTemplate() {
+        this.render('alternative_home');
+      }
+    }));
+
+    this.add('controller:myController', Controller.extend({
+      myValue: 'foo'
+    }));
+
+    this.addTemplate('alternative_home', '<p>alternative home: {{myValue}}</p>');
+
+
+    return this.visit('/').then(() => {
+      let homeRoute = this.applicationInstance.lookup('route:home');
+      let myController = this.applicationInstance.lookup('controller:myController');
+      let text = this.$('p').text();
+
+      assert.deepEqual(homeRoute.controller, myController,
+        'route controller is set by controllerName');
+
+      assert.equal(text, 'alternative home: foo',
+        'The homepage template was rendered with data from the custom controller');
+    });
+
+  }
+
+  [`@test The route controller specified via controllerName is used in render even when a controller with the routeName is available`](assert) {
+    this.router.map(function() {
+      this.route('home', { path: '/' });
+    });
+
+    this.addTemplate('home', '<p>home: {{myValue}}</p>');
+
+    this.add('route:home', Route.extend({
+      controllerName: 'myController'
+    }));
+
+    this.add('controller:home', Controller.extend({
+      myValue: 'home'
+    }));
+
+    this.add('controller:myController', Controller.extend({
+      myValue: 'myController'
+    }));
+
+    return this.visit('/').then(() => {
+      let homeRoute = this.applicationInstance.lookup('route:home');
+      let myController = this.applicationInstance.lookup('controller:myController');
+      let text = this.$('p').text();
+
+      assert.deepEqual(homeRoute.controller, myController,
+        'route controller is set by controllerName');
+
+      assert.equal(text, 'home: myController',
+        'The homepage template was rendered with data from the custom controller');
+    });
+  }
+
+  [`@test The Homepage with a 'setupController' hook modifying other controllers`](assert) {
+    this.router.map(function() {
+      this.route('home', { path: '/' });
+    });
+
+    this.add('route:home', Route.extend({
+      setupController(/* controller */) {
+        set(this.controllerFor('home'), 'hours', emberA([
+          'Monday through Friday: 9am to 5pm',
+          'Saturday: Noon to Midnight',
+          'Sunday: Noon to 6pm'
+        ]));
+      }
+    }));
+
+    this.addTemplate('home', '<ul>{{#each hours as |entry|}}<li>{{entry}}</li>{{/each}}</ul>');
+
+    return this.visit('/').then(() => {
+      let text = this.$('ul li:nth-child(3)').text();
+
+      assert.equal(text, 'Sunday: Noon to 6pm',
+        'The template was rendered with the hours context');
+    });
+  }
+
+  [`@test The Homepage with a computed context that does not get overridden`](assert) {
```

Comment:
please change the test to say `computed model that` instead of `context`, since `model` is the property with the CP.

Reply:
done